### PR TITLE
Added Maven goal to update pom.xml to latest Quarkus version

### PR DESCRIPTION
**Pull request by j1cken**
WDYT?

### CODE REVIEW

File: devtools/maven/src/main/java/io/quarkus/maven/UpdateToLatestVersionMojo.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        MavenXpp3Reader mavenreader = new MavenXpp3Reader();
+        Model model;
+        try {
+            FileReader reader = new FileReader(pomFile);
+            model = mavenreader.read(reader);
+            model.setPomFile(pomFile);
+        } catch (Exception e) {
+            throw new MojoExecutionException(e.getMessage());
+        }
+        MavenProject project = new MavenProject(model);
```
Comment (gastaldi) [inline devtools/maven/src/main/java/io/quarkus/maven/UpdateToLatestVersionMojo.java:68]: You  can inject the MavenProject directly in your mojo

**Comment by gastaldi** [comment]
The idea is good, but I'm afraid we'll need more than simply updating the version (but that's a good start). 

Also we should not forget our Gradle users that would need this feature too.

**Comment by gastaldi** [comment]
Also I'd recommend querying the Maven repositories for the latest Final version, not GitHub

File: devtools/maven/src/main/java/io/quarkus/maven/UpdateToLatestVersionMojo.java (hunk 1 above)
Comment (j1cken) [reply devtools/maven/src/main/java/io/quarkus/maven/UpdateToLatestVersionMojo.java:68]: I know, the idea was to have the existing POM file on disk without anything being resolved, transitively added and so on. I'm not 100% sure if this is the case if I inject it. I'll have a detailed look at that.

**Comment by j1cken** [comment]
> The idea is good, but I'm afraid we'll need more than simply updating the version (but that's a good start).

As stated it was a start, just to see if there's interest in such a feature before digging deeper and investing more time.
I was applying my "manual" steps here which are:
1. Generating a latest code example using code.quarkus.io
2. Copying over the properties section (that was enough so far) and usually only both the Quarkus versions changed
3. Rebuild and see if everything works

Surefire version has been one which I was paying attention to. But where do I get the version matrix (Quarkus <--> Surefire) from?

Any hints what else I should watch are very appreciated! 

> 
> Also we should not forget our Gradle users that would need this feature too.

Yep, on the radar already :) but as you said above I want to make sure to collect all requirements first. That's why I was starting this thread. Thx for commenting!

**Comment by j1cken** [comment]
> Also I'd recommend querying the Maven repositories for the latest Final version, not GitHub

Fully agree! Will change that. Even getting the Enterprise version "supported" makes it much easier!

**Comment by gastaldi** [comment]
> Surefire version has been one which I was paying attention to. But where do I get the version matrix (Quarkus <--> Surefire) from?

I think sticking to the latest Surefire works just fine. AFAIK the only reason why we haven't used `3.0.0-M5` is because isn't marked as stable yet.

**Comment by j1cken** [comment]
no need to implement anything, already there: https://www.mojohaus.org/versions-maven-plugin/

```mvn versions:update-properties```

**Pull request closed without merging**